### PR TITLE
Documented the incompatibility of shrink and cax kwargs in colorbar.

### PR DESCRIPTION
**Pull request by navdeeprana**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/colorbar.py [hunk 1]
```diff
@@ -180,6 +180,9 @@
 to which the colorbar is attached; but it is a manual method requiring
```
Comment (jklymak) [inline lib/matplotlib/colorbar.py]: I'd actually drop this part.  `*shtrink*` means to shrink for many reasons.  I routinely shrink my colorbars for esthetic reasons that have nothing to do w/ the size of the mappable.

File: lib/matplotlib/colorbar.py [hunk 2]
```diff
@@ -180,6 +180,9 @@
...
 some trial and error. If the colorbar is too tall (or a horizontal
 colorbar is too wide) use a smaller value of *shrink*.
+Further, *shrink* and *cax* kwargs are incompatible with each other and
```
Comment (jklymak) [inline lib/matplotlib/colorbar.py]: "Further, *shrink* " -> "The *shrink*".   Take or leave this, but "further" is kind of awkward unless really needed...

Comment (jklymak) [reply lib/matplotlib/colorbar.py]: I'd actually just replace this whole paragaph with: "Note that if *cax* is specified it determines the size of the colorbar and *shrink* and *aspect* kwargs are ignored"

If you are going to point this out for *shrink* it should also be pointed out for *aspect*.

Comment (navdeeprana) [reply lib/matplotlib/colorbar.py]: Replaced with "The *shrink* kwarg provides a simple way to scale the colorbar with respect
to the axes. Note that if *cax* is specified it determines the size of the
colorbar and *shrink* and *aspect* kwargs are ignored."

**Comment by NelleV** [comment]
Thanks @navdeeprana !

**Pull request merged**